### PR TITLE
Update build system

### PR DESCRIPTION
**Pull request by bisho**
* Use uv for venvs, build and publish (remove poetry)
* Use ruff for lint and format checks
* Simplify ci and release github pipelines
* Make nox use uv
* Clean nox, clean pyproject
* Update deps (meta-memcache-socket and others)

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    name: Publish to PyPI
+    runs-on: ubuntu-latest
+    # Environment and permissions trusted publishing.
+    environment:
+      name: production
+
+    permissions:
+      id-token: write
+    steps:
+      - uses: actions/checkout@v4
```
Comment (tonycosentini) [inline .github/workflows/release.yml:20]: Wow the release pipeline is so much simpler now

File: noxfile.py [hunk 2]
```diff
@@ -3,56 +3,58 @@
...
-nox.options.sessions = "lint", "types", "tests"
-locations = "src", "tests", "noxfile.py"
-DEFAULT_VERSION = "3.8"
-DEFAULT_BENCHMARK_VERSIONS = ["3.11"]
-VERSIONS = ["3.8", "3.11"]
+nox.options.sessions = "lint", "format", "types", "tests"
+locations = "src", "tests", "noxfile.py", "benchmark.py"
+DEFAULT_VERSION = "3.11"
+DEFAULT_BENCHMARK_VERSIONS = ["3.12"]
+VERSIONS = ["3.12", "3.11", "3.10"]
```
Comment (tonycosentini) [inline noxfile.py:10]: Is it worth adding 3.13 now that it's out?

File: src/meta_memcache/__init__.py [hunk 3]
```diff
@@ -1,4 +1,4 @@
-__version__ = "1.1.2"
+__version__ = "2.0.1"
```
Comment (tonycosentini) [inline src/meta_memcache/__init__.py:1]: Is there any way to get uv to inject this somehow? It seems easy to forget to change it.

**Review by tonycosentini — APPROVED** [review]

File: .github/workflows/release.yml (hunk 1 above)
Comment (bisho) [reply .github/workflows/release.yml:20]: And super fast... uv and ruff are magical :D

**Pull request merged**